pkg/cmd/server/kubernetes/master.go

```diff
@@ -171,5 +172,13 @@ func (c *MasterConfig) createSchedulerConfig() (*scheduler.Config, error) {
 	}
 
 	// if the config file isn't provided, use the default provider
-	return configFactory.CreateFromProvider(factory.DefaultProvider)
+	provider, err := factory.GetAlgorithmProvider(factory.DefaultProvider)
+	if err != nil {
+		return nil, err
+	}
+	// Register origin specific predicates and priority functions.
+	// TODO: Move generic predicates in origin to upstream k8s.
+	provider.FitPredicateKeys.Insert(originscheduler.RegisterOriginPredicates()...)
```

Comment:
Instead of doing this over here, it would be better if you modify the DefaultProvider in an init function within pkg/scheduler/<some-name>.go